### DISCUSSION TITLE
Make upload collections not require fields

### DISCUSSION THREAD

**Discussion by TomDo1234** [payloadcms/payload#1755] [Feature Requests & Ideas]
# Bug Report

## Steps to Reproduce

1. Create a collection
2. Make it an upload collection
3. Get TS error due to not having fields

I can fix this by creating a fields key with an empty array. But it would be nice and I guess neater if I don't need it if there is an upload field

**Comment by jmikrut** [comment]
Hey @TomDo1234 — this is a good call and I agree, although it will require some TS fanciness to only require `fields` when `upload` or `auth` is not enabled.

I will convert this to a feature request and we can keep track of it. For now, I think the empty array is the way to go.
